### ISSUE TITLE
can't run in docker on mac

### ISSUE THREAD

**Issue by wjljack** [giongto35/cloud-game#257]
make dev.run is ok 
```
I0106 16:20:09.181917   64470 main.go:30] Initializing coordinator server
I0106 16:20:09.182001   64470 main.go:31] Coordinator configs {{http://localhost:8000   {assets/games [] [neogeo pgm] true false} {6601 /coordinator false false} {8000 443  }} {1 {false 320 240} {{{assets/cores assets/cores} {true buildbot https://buildbot.libretro.com/nightly zip} map[gba:{mgba_libretro  [gba gbc] 240 160 0 false false false false} mame:{fbneo_libretro  [zip] 240 160 0 false false false false} n64:{mupen64plus_next_libretro mupen64plus_next_libretro.cfg [n64 v64 z64] 320 240 0 true true false false} nes:{nestopia_libretro  [nes] 256 240 0 false false false false} pcsx:{pcsx_rearmed_libretro pcsx_rearmed_libretro.cfg [cue] 350 240 0 false false false false} snes:{snes9x_libretro  [smc sfc swc fig bs] 256 224 0 false false true false}]}}} dev {[{stun:{server-ip}:3478  } {turn:{server-ip}:3478 root root}]}}
I0106 16:20:09.182606   64470 coordinator.go:47] Starting monitoring server for coordinator
I0106 16:20:09.183169   64470 game_library.go:141] [lib] scan... started
I0106 16:20:09.184136   64470 game_library.go:262] 
--------------------------------------------
--- The Library of ROMs                  ---
--------------------------------------------
    mslug (mslug.zip)
    Sample Demo by Florian (PD) (Sample Demo by Florian (PD).z64)
    Super Mario Bros (Super Mario Bros.nes)
    Sushi The Cat (Sushi The Cat.gba)
    anguna (anguna.gba)
--------------------------------------------
--- ROMs: 005                  214.787µs ---
--------------------------------------------
I0106 16:20:09.184149   64470 game_library.go:184] [lib] scan... completed
I0106 16:20:09.184154   64470 coordinator.go:113] Initializing Coordinator Server
I0106 16:20:09.188924   64471 worker.go:52] [worker] active modules: [monitoring::/worker:6601]
I0106 16:20:09.189134   64471 monitoring.go:40] [worker] Starting monitoring server at :6601
I0106 16:20:09.189162   64471 monitoring.go:62] [worker] Prometheus metric is enabled at :6601/worker/metrics
I0106 16:20:09.192968   64471 storage.go:26] Warn: Failed to create client: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
I0106 16:20:09.192995   64471 handlers.go:95] Starting Libretro cores sync...
I0106 16:20:09.193028   64471 handlers.go:117] Worker connecting to coordinator: ws://10.117.24.34:8000/wso?zone=
I0106 16:20:09.196160   64470 handlers.go:83] Coordinator: A worker is connecting...
I0106 16:20:09.196246   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Generated worker ID
I0106 16:20:09.196260   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Address:10.117.24.34
I0106 16:20:09.196293   64470 worker.go:34] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Is public: false zone: 
I0106 16:20:09.196297   64470 worker.go:34] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Set ping server address: http://localhost:9000/echo
I0106 16:20:09.196308   64471 handlers.go:72] [worker] connected to: 10.117.24.34:8000
I0106 16:20:09.235580   64471 internalhandlers.go:20] [worker] new id: a2b88b89-2963-4b30-95e1-423d3d81b239
I0106 16:20:12.473862   64470 handlers.go:152] Coordinator: A user is connecting...
I0106 16:20:12.473930   64470 browser.go:33] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Generated worker ID
I0106 16:20:12.473950   64470 browser.go:29] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Get Room  Zone  From URL /ws?room_id=&zone=
I0106 16:20:12.473991   64470 browser.go:33] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Send sync[http://localhost:9000/echo]http://localhost:9000/echo
I0106 16:20:12.537505   64471 server.go:33] &{0xc0003d06e0 0xc0001a6700 {} 0x4096e00 false false false false 0xc0003c6600 {0xc0003e60e0 map[] false false} map[Access-Control-Allow-Origin:[*]] true 0 -1 0 false false [] 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [0 0 0 0 0 0 0 0 0 0] [0 0 0] 0xc000036460 0} echo
I0106 16:20:12.558731   64470 browser.go:33] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Latency mapmap[0xc000198d80:27]
I0106 16:20:12.562728   64470 browser.go:29] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Received init_webrtc request -> relay to worker: a2b88b89-2963-4b30-95e1-423d3d81b239
I0106 16:20:12.923882   64471 internalhandlers.go:46] Received a request to createOffer from browser via coordinator
I0106 16:20:12.923966   64471 webrtc.go:121] === StartClient ===
I0106 16:20:12.934892   64471 webrtc.go:143] Add video track
I0106 16:20:12.937506   64471 webrtc.go:233] Created Offer
I0106 16:20:12.938083   64471 internalhandlers.go:74] Start peerconnection 7cf10868-119e-4118-92d7-3395c9e21255
I0106 16:20:12.938217   64471 webrtc.go:194] OnIceCandidate: candidate:4112801032 1 udp 2130706431 2408:864e:c02:54:1096:b83c:a8e8:73dd 58298 typ host
I0106 16:20:12.938294   64471 webrtc.go:194] OnIceCandidate: candidate:281475904 1 udp 2130706431 2408:864e:c02:54:b48d:45b2:cfa:62be 51240 typ host
I0106 16:20:12.938429   64470 browser.go:33] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Received SDP from worker -> sending back to browser
I0106 16:20:12.938476   64471 webrtc.go:194] OnIceCandidate: candidate:314150723 1 udp 2130706431 2408:864e:c02:54:5682:e9bc:8ba1:b0e2 63964 typ host
I0106 16:20:12.938522   64471 webrtc.go:194] OnIceCandidate: candidate:1694255302 1 udp 2130706431 10.117.24.34 59752 typ host
I0106 16:20:12.938812   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Received IceCandidate from worker -> relay to browser
I0106 16:20:13.013001   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Received IceCandidate from worker -> relay to browser
I0106 16:20:13.013039   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Received IceCandidate from worker -> relay to browser
I0106 16:20:13.013068   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Received IceCandidate from worker -> relay to browser
I0106 16:20:13.013098   64470 browser.go:33] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Received browser answered SDP -> relay to worker
I0106 16:20:13.013127   64470 browser.go:33] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Received IceCandidate from browser -> relay to worker
I0106 16:20:13.013150   64470 browser.go:33] Browser 7cf10868-119e-4118-92d7-3395c9e21255] Received IceCandidate from browser -> relay to worker
I0106 16:20:13.013482   64471 internalhandlers.go:108] Received remote Ice Candidate from browser
I0106 16:20:13.013498   64471 internalhandlers.go:108] Received remote Ice Candidate from browser
I0106 16:20:13.016596   64471 internalhandlers.go:90] Received answer SDP from browser
I0106 16:20:13.016650   64471 webrtc.go:277] Decoded Ice: candidate:1718747246 1 udp 2113939711 c74471a1-aa42-4d4d-8066-ace148ac9739.local 51080 typ host generation 0 ufrag KMYE network-cost 999
I0106 16:20:13.016666   64471 webrtc.go:281] Add Ice candidate from peer failed
I0106 16:20:13.016706   64471 webrtc.go:277] Decoded Ice: candidate:2986062645 1 udp 2113937151 36da5bc6-0861-4484-830f-c58771ba781d.local 54002 typ host generation 0 ufrag KMYE network-cost 999
I0106 16:20:13.016724   64471 webrtc.go:281] Add Ice candidate from peer failed
I0106 16:20:13.016739   64471 internalhandlers.go:116] Error: Cannot add IceCandidate of client: 7cf10868-119e-4118-92d7-3395c9e21255
I0106 16:20:13.016749   64471 internalhandlers.go:116] Error: Cannot add IceCandidate of client: 7cf10868-119e-4118-92d7-3395c9e21255
I0106 16:20:13.018116   64471 webrtc.go:266] Set Remote Description
I0106 16:20:13.018187   64471 webrtc.go:178] ICE Connection State has changed: checking
I0106 16:20:13.028832   64471 webrtc.go:194] OnIceCandidate: candidate:1812110816 1 udp 1694498815 220.249.79.114 6867 typ srflx raddr 0.0.0.0 rport 60244
I0106 16:20:13.029004   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Received IceCandidate from worker -> relay to browser
I0106 16:20:13.029018   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Received IceCandidate from worker -> relay to browser
I0106 16:20:14.085100   64471 webrtc.go:178] ICE Connection State has changed: connected
I0106 16:20:14.085537   64471 webrtc.go:182] ConnectionStateConnected
I0106 16:20:14.085546   64471 webrtc.go:317] Start streaming
I0106 16:20:14.090928   64471 webrtc.go:162] Data channel 'game-input'-'824641596248' open.
^CI0106 16:36:45.786460   64471 monitoring.go:72] [worker] Shutting down monitoring server
I0106 16:36:45.789658   64470 cws.go:178] [!] read: websocket: close 1006 (abnormal closure): unexpected EOF
I0106 16:36:45.789679   64470 worker.go:38] Worker a2b88b89-2963-4b30-95e1-423d3d81b239] Unregister worker from coordinator
```

but make dev.run-docker
```
Attaching to cloud-game-local
cloud-game-local | (EE) 
cloud-game-local | Fatal server error:
cloud-game-local | (EE) Server is already active for display 99
cloud-game-local |      If this server is no longer running, remove /tmp/.X99-lock
cloud-game-local |      and start again.
cloud-game-local | (EE) 
cloud-game-local | I0106 08:37:13.066371       9 main.go:30] Initializing coordinator server
cloud-game-local | I0106 08:37:13.066468       9 main.go:31] Coordinator configs {{http://localhost:8000   {assets/games [] [neogeo pgm] true false} {6601 /coordinator false false} {8000 443  }} {1 {false 320 240} {{{assets/cores assets/cores} {true buildbot https://buildbot.libretro.com/nightly zip} map[gba:{mgba_libretro  [gba gbc] 240 160 0 false false false false} mame:{fbneo_libretro  [zip] 240 160 0 false false false false} n64:{mupen64plus_next_libretro mupen64plus_next_libretro.cfg [n64 v64 z64] 320 240 0 true true false false} nes:{nestopia_libretro  [nes] 256 240 0 false false false false} pcsx:{pcsx_rearmed_libretro pcsx_rearmed_libretro.cfg [cue] 350 240 0 false false false false} snes:{snes9x_libretro  [smc sfc swc fig bs] 256 224 0 false false true false}]}}} dev {[{stun:{server-ip}:3478  } {turn:{server-ip}:3478 root root}]}}
cloud-game-local | I0106 08:37:13.066747       9 game_library.go:141] [lib] scan... started
cloud-game-local | I0106 08:37:13.067121       9 coordinator.go:47] Starting monitoring server for coordinator
cloud-game-local | I0106 08:37:13.067261      10 worker.go:52] [worker] active modules: [monitoring::/worker:6601]
cloud-game-local | I0106 08:37:13.067527      10 monitoring.go:40] [worker] Starting monitoring server at :6601
cloud-game-local | I0106 08:37:13.067566      10 monitoring.go:62] [worker] Prometheus metric is enabled at :6601/worker/metrics
cloud-game-local | I0106 08:37:13.075619       9 game_library.go:262] 
cloud-game-local | --------------------------------------------
cloud-game-local | --- The Library of ROMs                  ---
cloud-game-local | --------------------------------------------
cloud-game-local |     Sample Demo by Florian (PD) (Sample Demo by Florian (PD).z64)
cloud-game-local |     Super Mario Bros (Super Mario Bros.nes)
cloud-game-local |     Sushi The Cat (Sushi The Cat.gba)
cloud-game-local |     anguna (anguna.gba)
cloud-game-local |     mslug (mslug.zip)
cloud-game-local | --------------------------------------------
cloud-game-local | --- ROMs: 005                   8.7667ms ---
cloud-game-local | --------------------------------------------
cloud-game-local | I0106 08:37:13.075769       9 game_library.go:184] [lib] scan... completed
cloud-game-local | I0106 08:37:13.075817       9 coordinator.go:113] Initializing Coordinator Server
cloud-game-local | I0106 08:37:13.076482      10 storage.go:26] Warn: Failed to create client: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
cloud-game-local | I0106 08:37:13.076540      10 handlers.go:95] Starting Libretro cores sync...
cloud-game-local | I0106 08:37:13.076798      10 handlers.go:117] Worker connecting to coordinator: ws://10.117.24.34:8000/wso?zone=
cloud-game-local | I0106 08:37:13.078853       9 handlers.go:83] Coordinator: A worker is connecting...
cloud-game-local | I0106 08:37:13.079051       9 worker.go:38] Worker e409333a-2029-4aff-83f9-8a5ea06f3d3a] Generated worker ID
cloud-game-local | I0106 08:37:13.079078       9 worker.go:38] Worker e409333a-2029-4aff-83f9-8a5ea06f3d3a] Address:172.22.0.1
cloud-game-local | I0106 08:37:13.079222       9 worker.go:34] Worker e409333a-2029-4aff-83f9-8a5ea06f3d3a] Is public: false zone: 
cloud-game-local | I0106 08:37:13.079328       9 worker.go:34] Worker e409333a-2029-4aff-83f9-8a5ea06f3d3a] Set ping server address: http://localhost:9000/echo
cloud-game-local | I0106 08:37:13.079682      10 handlers.go:72] [worker] connected to: 10.117.24.34:8000
cloud-game-local | I0106 08:37:13.080317      10 internalhandlers.go:20] [worker] new id: e409333a-2029-4aff-83f9-8a5ea06f3d3a
cloud-game-local | I0106 08:37:19.227657       9 handlers.go:152] Coordinator: A user is connecting...
cloud-game-local | I0106 08:37:19.227775       9 browser.go:33] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Generated worker ID
cloud-game-local | I0106 08:37:19.227828       9 browser.go:29] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Get Room  Zone  From URL /ws?room_id=&zone=
cloud-game-local | I0106 08:37:19.227893       9 browser.go:33] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Send sync[http://localhost:9000/echo]http://localhost:9000/echo
cloud-game-local | I0106 08:37:19.274758      10 server.go:33] &{0xc00026c320 0xc00037a000 {} 0x4f3440 false false false false 0 {0 0} 0xc000314a00 {0xc0001281c0 map[] false false} map[Access-Control-Allow-Origin:[*]] true 0 -1 0 false false [] 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [0 0 0 0 0 0 0 0 0 0] [0 0 0] 0xc00007e230 0} echo
cloud-game-local | I0106 08:37:19.290372       9 browser.go:33] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Latency mapmap[0xc000200360:19]
cloud-game-local | I0106 08:37:19.296020       9 browser.go:29] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Received init_webrtc request -> relay to worker: e409333a-2029-4aff-83f9-8a5ea06f3d3a
cloud-game-local | I0106 08:37:19.297378      10 internalhandlers.go:46] Received a request to createOffer from browser via coordinator
cloud-game-local | I0106 08:37:19.297545      10 webrtc.go:121] === StartClient ===
cloud-game-local | I0106 08:37:19.305726      10 webrtc.go:143] Add video track
cloud-game-local | I0106 08:37:19.306666      10 webrtc.go:233] Created Offer
cloud-game-local | I0106 08:37:19.306923      10 internalhandlers.go:74] Start peerconnection 8008fba7-6464-4853-b02a-6a250f5876f1
cloud-game-local | I0106 08:37:19.308077      10 webrtc.go:194] OnIceCandidate: candidate:3958270407 1 udp 2130706431 172.22.0.2 49848 typ host
cloud-game-local | I0106 08:37:19.308485       9 browser.go:33] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Received SDP from worker -> sending back to browser
cloud-game-local | I0106 08:37:19.309788       9 worker.go:38] Worker e409333a-2029-4aff-83f9-8a5ea06f3d3a] Received IceCandidate from worker -> relay to browser
cloud-game-local | I0106 08:37:19.349715       9 browser.go:33] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Received browser answered SDP -> relay to worker
cloud-game-local | I0106 08:37:19.351147      10 internalhandlers.go:90] Received answer SDP from browser
cloud-game-local | I0106 08:37:19.352487      10 webrtc.go:266] Set Remote Description
cloud-game-local | I0106 08:37:19.352823      10 webrtc.go:178] ICE Connection State has changed: checking
cloud-game-local | I0106 08:37:19.361924       9 browser.go:33] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Received IceCandidate from browser -> relay to worker
cloud-game-local | I0106 08:37:19.363034       9 browser.go:33] Browser 8008fba7-6464-4853-b02a-6a250f5876f1] Received IceCandidate from browser -> relay to worker
cloud-game-local | I0106 08:37:19.363287      10 internalhandlers.go:108] Received remote Ice Candidate from browser
cloud-game-local | I0106 08:37:19.363343      10 webrtc.go:277] Decoded Ice: candidate:2986062645 1 udp 2113937151 0714db85-1721-484a-b894-c756bc164a50.local 51221 typ host generation 0 ufrag CR9F network-cost 999
cloud-game-local | I0106 08:37:19.363449      10 webrtc.go:285] Add Ice Candidate: candidate:2986062645 1 udp 2113937151 0714db85-1721-484a-b894-c756bc164a50.local 51221 typ host generation 0 ufrag CR9F network-cost 999
cloud-game-local | I0106 08:37:19.364526      10 internalhandlers.go:108] Received remote Ice Candidate from browser
cloud-game-local | I0106 08:37:19.364695      10 webrtc.go:277] Decoded Ice: candidate:1718747246 1 udp 2113939711 7c45db69-c4f4-4f82-961a-98864a0df664.local 49247 typ host generation 0 ufrag CR9F network-cost 999
cloud-game-local | I0106 08:37:19.364868      10 webrtc.go:285] Add Ice Candidate: candidate:1718747246 1 udp 2113939711 7c45db69-c4f4-4f82-961a-98864a0df664.local 49247 typ host generation 0 ufrag CR9F network-cost 999
cloud-game-local | I0106 08:37:19.549324      10 webrtc.go:194] OnIceCandidate: candidate:1812110816 1 udp 1694498815 220.249.79.114 19460 typ srflx raddr 0.0.0.0 rport 36912
cloud-game-local | I0106 08:37:19.550453       9 worker.go:38] Worker e409333a-2029-4aff-83f9-8a5ea06f3d3a] Received IceCandidate from worker -> relay to browser
cloud-game-local | I0106 08:37:19.550612       9 worker.go:38] Worker e409333a-2029-4aff-83f9-8a5ea06f3d3a] Received IceCandidate from worker -> relay to browser
cloud-game-local | I0106 08:37:49.399432      10 webrtc.go:178] ICE Connection State has changed: failed
```

**Comment by wjljack** [comment]
when run in docker 
```
cloud-game-local | I0106 08:37:19.308077      10 webrtc.go:194] OnIceCandidate: candidate:3958270407 1 udp 2130706431 172.22.0.2 49848 typ host
```
udp ip address is container ip `172.22.0.2`

**Comment by wjljack** [comment]
and if i use `    network_mode: "host"`  in compose , server is running but none of ports available

**Comment by sergystepanov** [comment]
@wjljack, thanks for the report. Unfortunately, I don't have a Mac to try to test it but on Windows (without WSL2), it doesn't work either. So I'll be guessing. In short: we need "host" network mode because WebRTC will need a wide range of randomly opened ports (that's why your first attempt won't work), next there's a [problem](https://docs.docker.com/docker-for-mac/networking/) with the "host" mode on both macOS and Windows it seems:
```
There is no docker0 bridge on macOS
Because of the way networking is implemented in Docker Desktop for Mac, 
you cannot see a docker0 interface on the host. 
This interface is actually within the virtual machine.

I cannot ping my containers
Docker Desktop for Mac can’t route traffic to containers.
```
which means that in that mode only the container can access your host network and not vice versa.
Personally, I don't know much about Docker networking yet, but will try to search for a solution, maybe there is one, or if you are interested, you could try it as well.
To sum it up, it won't work in Docker for Mac and Windows. But should work natively.

**Comment by wjljack** [comment]
> @wjljack, thanks for the report. Unfortunately, I don't have a Mac to try to test it but on Windows (without WSL2), it doesn't work either. So I'll be guessing. In short: we need "host" network mode because WebRTC will need a wide range of randomly opened ports (that's why your first attempt won't work), next there's a [problem](https://docs.docker.com/docker-for-mac/networking/) with the "host" mode on both macOS and Windows it seems:
> 
> ```
> There is no docker0 bridge on macOS
> Because of the way networking is implemented in Docker Desktop for Mac, 
> you cannot see a docker0 interface on the host. 
> This interface is actually within the virtual machine.
> 
> I cannot ping my containers
> Docker Desktop for Mac can’t route traffic to containers.
> ```
> 
> which means that in that mode only the container can access your host network and not vice versa.
> Personally, I don't know much about Docker networking yet, but will try to search for a solution, maybe there is one, or if you are interested, you could try it as well.
> To sum it up, it won't work in Docker for Mac and Windows. But should work natively.

thx,i will try it on native linux

**Comment by giongto35** [comment]
Thanks @sergystepanov  and @wjljack for reporting this issue.
It works on my Mac before. The project should be able to run on Mac OS. Let me recheck.

**Comment by sergystepanov** [comment]
@giongto35, that'll be great if host mode works.
It's strange though because it shouldn't work if Docker [documentation](https://docs.docker.com/network/network-tutorial-host/#prerequisites) is correct:
> The `host` networking driver only works on Linux hosts, and is not supported on Docker Desktop for Mac, Docker Desktop for Windows, or Docker EE for Windows Server.

I tried some hacking with bridged network, found how to limit random port range in Pion but ICE discovers candidates only with Docker internal IP: 172.17.x.x

```
	settingEngine := SettingEngine{}
	settingEngine.SetEphemeralUDPPortRange(50000, 50005)
	// settingEngine.SetNAT1To1IPs([]string{"95.107.38.237"}, ICECandidateTypeHost)

	api := NewAPI(WithMediaEngine(m), WithInterceptorRegistry(i), WithSettingEngine(settingEngine))
	return api.NewPeerConnection(conf)

// docker-compose
    ports:
      - "8000:8000"
      - "9000:9000"
      - "3478:3478/tcp"
      - "3478:3478/udp"
      - "50000-50005:50000-50005/udp"
```

**Comment by sergystepanov** [comment]
lol, it works in bridged mode with IP replacement to 127.0.0.1 as I guessed `settingEngine.SetNAT1To1IPs([]string{"127.0.0.1"}, ICECandidateTypeHost)`. Will add into conf these params and make a PR to test.

![image](https://user-images.githubusercontent.com/846874/103903281-c0bcfb00-510c-11eb-9443-a8d03d157f8d.png)

**Comment by sergystepanov** [comment]
@wjljack, it seems the previous master code should work in macOS without changes.
Make sure that you are using the latest code snapshot, also you can try to reinstall Docker, check your firewalls, maybe it blocks the connections.
If it won't help then try this docker-compose.yml config with my changes:
```
version: '3'
services:

  cloud-game:
    build: .
    image: cloud-game-local
    container_name: cloud-game-local
    privileged: true
    environment:
      - DISPLAY=:99
      - MESA_GL_VERSION_OVERRIDE=3.3
      - CLOUD_GAME_WEBRTC_ICEPORTS_MIN=50000
      - CLOUD_GAME_WEBRTC_ICEPORTS_MAX=50010
      - CLOUD_GAME_WEBRTC_ICEIPMAP=127.0.0.1
    # network_mode: "host"
    command: >
      bash -c "Xvfb :99 & coordinator --v=5 & worker --coordinatorhost localhost:8000"
    ports:
      - "8000:8000"
      - "9000:9000"
      - "50000-50010:50000-50010/udp"
    volumes:
      # keep cores persistent in the cloud-game_cores volume
      - cores:/usr/local/share/cloud-game/assets/cores
      - ${CLOUD_GAME_GAMES_PATH}:/usr/local/share/cloud-game/assets/games

volumes:
  cores:
```
In this config, Docker will open 11 ports from MIN-MAX range, and IP for server connections will be set to 127.0.0.1 the address where these ports are forwarded from Docker -- to your host which is localhost.
I Hope, this helps.
